### PR TITLE
Fix dead test code

### LINKED ISSUE #53415: Dead conditional code in test
See https://github.com/dotnet/roslyn/pull/53413. The test still passes.

The condition under `compilation.Assembly.RuntimeSupportsCovariantReturnsOfClasses` is unreachable.

- There are two tests in the same test file doing the same thing.
- Possibly the same issue in a test in `CodeGenOverridingAndHiding`.
- Possibly the same issue in two tests in `SymbolErrorTests` (they check `RuntimeSupportsDefaultInterfaceImplementation`)

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #53415

The `Assembly.RuntimeSupports*` was never true in these tests. Ran them against both TFMs so both code paths are actually tested.

### CODE REVIEW

File: src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenOverridingAndHiding.cs [hunk 1]
```diff
@@ -1114,8 +1114,10 @@ public static void Main()
             var comp = CompileAndVerify(source, expectedOutput: @"2545571191011111114151617");
         }
 
-        [Fact]
-        public void TestAmbiguousOverridesRefOut()
+        [Theory]
+        [InlineData(TargetFramework.NetCoreApp)]
+        [InlineData(TargetFramework.Standard)]
+        public void TestAmbiguousOverridesRefOut(TargetFramework targetFramework)
```
Comment (Youssef1313) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenOverridingAndHiding.cs:1120]: This test fails with:

```
  Failed Microsoft.CodeAnalysis.CSharp.UnitTests.CodeGen.CodeGenOverridingAndHidingTests.TestAmbiguousOverridesRefOut(targetFramework: NetCoreApp) [47 ms]
  Error Message:
   System.Exception : Verification failed
---- Roslyn.Test.Utilities.Desktop.RuntimePeVerifyException : 
PeVerify failed for assembly '':
Type load failed.
Type load failed.
Type load failed.
Type load failed.


  Stack Trace:
     at Roslyn.Test.Utilities.Desktop.DesktopRuntimeEnvironment.Verify(Verification verification) in /_/src/Compilers/Test/Core/Platform/Desktop/DesktopRuntimeEnvironment.cs:line 324
   at Microsoft.CodeAnalysis.Test.Utilities.CompilationVerifier.Emit(String expectedOutput, Nullable`1 expectedReturnCode, String[] args, IEnumerable`1 manifestResources, EmitOptions emitOptions, Verification peVerify, SignatureDescription[] expectedSignatures) in /_/src/Compilers/Test/Core/CompilationVerifier.cs:line 205
   at Microsoft.CodeAnalysis.Test.Utilities.CommonTestBase.Emit(Compilation compilation, IEnumerable`1 dependencies, IEnumerable`1 manifestResources, SignatureDescription[] expectedSignatures, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, Action`1 assemblyValidator, Action`1 symbolValidator, EmitOptions emitOptions, Verification verify) in /_/src/Compilers/Test/Core/CommonTestBase.cs:line 153
   at Microsoft.CodeAnalysis.Test.Utilities.CommonTestBase.CompileAndVerifyCommon(Compilation compilation, IEnumerable`1 manifestResources, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 assemblyValidator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, EmitOptions emitOptions, Verification verify) in /_/src/Compilers/Test/Core/CommonTestBase.cs:line 67
   at Microsoft.CodeAnalysis.CSharp.Test.Utilities.CSharpTestBase.CompileAndVerify(Compilation compilation, IEnumerable`1 manifestResources, IEnumerable`1 dependencies, Action`1 sourceSymbolValidator, Action`1 validator, Action`1 symbolValidator, SignatureDescription[] expectedSignatures, String expectedOutput, Nullable`1 expectedReturnCode, String[] args, EmitOptions emitOptions, Verification verify) in /_/src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs:line 842
   at Microsoft.CodeAnalysis.CSharp.UnitTests.CodeGen.CodeGenOverridingAndHidingTests.TestAmbiguousOverridesRefOut(TargetFramework targetFramework) in /_/src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenOverridingAndHiding.cs:line 1264
----- Inner Stack Trace -----
   at Roslyn.Test.Utilities.Desktop.RuntimeAssemblyManager.PeVerifyModules(String[] modulesToVerify, Boolean throwOnError) in /_/src/Compilers/Test/Core/Platform/Desktop/RuntimeAssemblyManager.cs:line 459
   at Roslyn.Test.Utilities.Desktop.RuntimeAssemblyManager.PeVerifyModules(String[] modulesToVerify, Boolean throwOnError)
   at Roslyn.Test.Utilities.Desktop.DesktopRuntimeEnvironment.Verify(Verification verification) in /_/src/Compilers/Test/Core/Platform/Desktop/DesktopRuntimeEnvironment.cs:line 314
Results File: C:\h\w\ACF40984\w\A71F0957\e\Microsoft.CodeAnalysis.CSharp.Emit.UnitTests.dll.8_net472_x86_test_results.xml

```

Should I skip verification here?

File: src/Compilers/CSharp/Test/Semantic/Semantics/InheritanceBindingTests.cs [hunk 2]
```diff
@@ -4922,8 +4922,10 @@ public class Derived2 : Derived
             CreateCompilation(text).VerifyDiagnostics();
         }
 
-        [Fact]
-        public void TestAmbiguousOverrideMethod()
+        [Theory]
+        [InlineData(TargetFramework.NetCoreApp)]
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/InheritanceBindingTests.cs:4926]: >NetCoreApp [](http://example.com/codeflow?start=36&length=10)

I think we should instead always use ```TargetFramework.StandardLatest``` for CreateCompilation

**Comment by AlekseyTs** [comment]
Done with review pass (commit 1)

**Comment by Youssef1313** [comment]
Thanks @AlekseyTs! This is much better now :) 🎉

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 2)

**Comment by AlekseyTs** [comment]
@Youssef1313 Thanks for the contribution.

**Pull request merged**